app/questionnaire/placeholder_parser.py

```diff
@@ -96,23 +82,39 @@ def _parse_transforms(self, transform_list: Sequence[Mapping]):
         transformed_value = None
 
         for transform in transform_list:
-            transform_args: Dict[str, Union[None, str, List[str]]] = {}
+            transform_args: Dict[
+                str, Union[None, str, int, Decimal, dict, list[str]]
+            ] = {}
+
             for arg_key, arg_value in transform["arguments"].items():
-                if not isinstance(arg_value, dict):
-                    transform_args[arg_key] = arg_value
-                elif "value" in arg_value:
-                    transform_args[arg_key] = arg_value["value"]
-                elif arg_value["source"] == "previous_transform":
-                    transform_args[arg_key] = transformed_value
+                if isinstance(arg_value, list):
+                    transform_args[arg_key] = self._resolve_value_source_list(arg_value)
+                elif isinstance(arg_value, dict):
+                    if "value" in arg_value:
+                        transform_args[arg_key] = arg_value["value"]
+                    elif arg_value["source"] == "previous_transform":
+                        transform_args[arg_key] = transformed_value
+                    else:
+                        transform_args[arg_key] = self._resolve_value_source(arg_value)
                 else:
-                    transform_args[arg_key] = self._resolve_value_source(arg_value)
+                    transform_args[arg_key] = arg_value
 
             transformed_value = getattr(self._transformer, transform["transform"])(
                 **transform_args
             )
 
         return transformed_value
 
+    def _resolve_value_source_list(self, value_source_list: List[Dict]) -> list[str]:
```

Comment:
The built-in types should be used.
```suggestion
    def _resolve_value_source_list(self, value_source_list: list[dict]) -> list[str]:
```

I think all types in this file should be changed to use the built-in types.